### ISSUE TITLE
probabilities_ outcome not consistent with get_document_info output

### ISSUE THREAD

**Issue by chentitus** [MaartenGr/BERTopic#2017]
Hi Maarten,

I found that `probabilities_` outcome is not always the same with the outcome of `get_document_info`.

`topic_model.probabilities_[0]` gave me the following:

`array([0.31045925, 0.0031012 , 0.00162018, 0.00349846, 0.0039572 ,
       0.01974048, 0.00788453, 0.00418387, 0.00252376, 0.00240863,
       0.00308729, 0.00263459, 0.00250224, 0.00254411, 0.00254329,
       0.00283324, 0.00229805, 0.00638228, 0.00281222, 0.0028788 ,
       0.00298958, 0.0030663 , 0.00246619, 0.00387198, 0.00227671,
       0.00254577, 0.00245718, 0.0026639 , 0.00243602, 0.00419228,
       0.00243791, 0.00244527, 0.00384736, 0.00285853, 0.00214743,
       0.00218918, 0.0029944 ])`

From this output it seems that the dominant topic would be Topic 0.

However, `get_document_info` gave me this:

Document                   在中国 嫦娥五号 探月工程 顺利实施 之际 美国 在航天 领域 频频 动作 退出_开放天空条...  
Topic                                                                      5
Name                                                      5_太空_卫星_外空_外空_攻防对抗
Representation             [太空, 卫星, 外空, 外空_攻防对抗, 外空_攻防对抗 准备, 外空安全, 发射, 航天...
Representative_Docs        [进入 世纪 以来 国际社会 围绕 外层空间 争夺 日益 激烈 外空安全 形势 日趋_紧张 ...
Top_n_words                太空 - 卫星 - 外空 - 外空_攻防对抗 - 外空_攻防对抗 准备 - 外空安全 - 发...
Probability                                                         0.310459
Representative_document                                                False
Name: 0, dtype: object

This output shows the same highest `Probability` (which is 0.310459) but assigns the document to Topic 5 instead of Topic 0.

Any solutions to this issue? Thanks much for the help!

**Comment by MaartenGr** [comment]
There are a couple of issues open about exactly this so I would advise looking there. However, to get started it is important to understand how the probabilities (assuming you use HDBSCAN) are created. If you check the documentation of HDBSCAN ([here](https://hdbscan.readthedocs.io/en/latest/soft_clustering.html) and [here](https://hdbscan.readthedocs.io/en/latest/soft_clustering_explanation.html)) you might notice that the probabilities are generated **after** the actual cluster assignment. This means that the soft clustering, and therefore probability generation, is an *approximation* rather than part of the inherent training/assignment process. As a result, these indeed might be different. 

Having said that, it might also relate to the code that you are using, especially if you use a model different from HDBSCAN. Could you share the standard info, like the code, BERTopic version, etc? That makes it easier to understand how the probabilities are calculated (there are different methods implemented).

**Comment by chentitus** [comment]
Sure, here comes my code:

`data = pd.read_csv("space_para.csv")
sampled_data2 = data['2grams_filtered'].tolist() `

` from sentence_transformers import SentenceTransformer `
` from bertopic import BERTopic `
` import numpy as np `
` from scipy.sparse import csr_matrix `
` from umap import UMAP `
` from hdbscan import HDBSCAN `

` sentence_model = SentenceTransformer("paraphrase-multilingual-MiniLM-L12-v2", device="mps") `
` embeddings = sentence_model.encode(sampled_data2, show_progress_bar=True) `

` np.random.seed(42) `

` umap_model = UMAP(n_neighbors=200, n_components=10, min_dist=0.1, metric='cosine', random_state=42) `

` hdbscan_model = HDBSCAN(min_cluster_size=20, metric='euclidean', cluster_selection_method='eom', prediction_data=True,
                        min_samples=5) `

` from sklearn.feature_extraction.text import CountVectorizer `

` vectorizer_model = CountVectorizer(ngram_range=(1, 2), stop_words=stopWord) `

` from bertopic.vectorizers import ClassTfidfTransformer `
` ctfidf_model = ClassTfidfTransformer(seed_words=['太空','卫星'], bm25_weighting=True, reduce_frequent_words=True) `

` topic_model = BERTopic(embedding_model="distiluse-base-multilingual-cased-v1", language="multilingual",
                       verbose=True,
                       calculate_probabilities=True,
                       umap_model=umap_model,
                       hdbscan_model=hdbscan_model,
                       vectorizer_model=vectorizer_model,
                       ctfidf_model=ctfidf_model,
                       nr_topics="auto",
                       n_gram_range=(1, 2),
                       top_n_words=15,
                       min_topic_size=250) `

` topics, probs = topic_model.fit_transform(sampled_data2, embeddings) `
` topic_model.probabilities_ `
` topic_model.get_topic_info() `

**Comment by MaartenGr** [comment]
Seeing as you use HDBSCAN it is indeed as a result of what I mentioned previously. Do also note that since you reduced topics with `nr_topics="auto"` the probabilities get merged. Note that per the [best practices](https://maartengr.github.io/BERTopic/getting_started/best_practices/best_practices.html#controlling-number-of-topics) I would advise using primarily `min_topic_size` instead to control the number of topics.
